### PR TITLE
Wallet: create forgot password page

### PR DESCRIPTION
**Pull request by fmtabbara**
# Description

Creates a new page for a user to "reset" their password

Closes:
https://github.com/nymtech/nym/issues/2409
https://github.com/nymtech/nym/issues/2396

# Checklist:

- [ ] added a changelog entry to `CHANGELOG.md`

### CODE REVIEW

File: nym-wallet/src/pages/auth/pages/signin-mnemonic.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ export const SignInMnemonic = () => {
     setPasswordExists(hasPassword);
   };
 
-  const handlePageChange = (page: string) => {
+  const handlePageChange = (page: any) => {
```
Comment (mmsinclair) [inline nym-wallet/src/pages/auth/pages/signin-mnemonic.tsx:21]:
```suggestion
  const handlePageChange = (page: string | number) => {
```

File: nym-wallet/src/pages/auth/pages/forgot-password.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      <Stack spacing={2} sx={{ width: 300 }}>
+        <Button
+          variant="contained"
+          size="large"
+          onClick={async () => {
+            try {
+              await archiveWalletFile();
+            } catch (e) {
+              Console.error(e);
+              enqueueSnackbar('Failed to archive wallet file', { variant: 'error' });
```
Comment (mmsinclair) [inline nym-wallet/src/pages/auth/pages/forgot-password.tsx]:
```suggestion
              enqueueSnackbar('Failed to archive your existing wallet file', { variant: 'error' });
```

**Review by mmsinclair — CHANGES_REQUESTED** [review]
Looks good - two minor changes

File: nym-wallet/src/pages/auth/pages/signin-mnemonic.tsx (hunk 1 above)
Comment (fmtabbara) [reply nym-wallet/src/pages/auth/pages/signin-mnemonic.tsx:21]: I did try this before but I get a type error. I think we'll need to add the react router types package to get around this

**Review by mmsinclair — APPROVED** [review]

**Pull request merged**